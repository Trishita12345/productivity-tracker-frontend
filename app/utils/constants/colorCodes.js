export const COLORS = {
  LIGHT_GREY: '#F5F5F5',
  GREY: '#C6C6C6',
  DARK_GREY: '#87797F',
  VIOLET: '#70367C',
  ORANGE: '#FD7E50',
  SKY: '#95D0D5',
  DARK_VIOLET: '#301137',
  GREEN: '#6DB33F',
  YELLOW: '#DCD41D',
  RED: '#E41515',
  WHITE: '#FFFAFF',
  BLUE: '#4E63B2',
};
