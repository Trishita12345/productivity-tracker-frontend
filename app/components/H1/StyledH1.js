import styled from 'styled-components';
import { COLORS } from '../../utils/constants/colorCodes';

const textColor = {
  primary: COLORS.DARK_VIOLET,
  secondary: COLORS.WHITE,
  tertiary: COLORS.DARK_GREY,
};

const StyledH1 = styled.h1`
  font-family: 'Roboto Slab';
  font-style: normal;
  font-weight: bold;
  font-size: 2em;
  line-height: 24px;
  color: ${({ variant }) => textColor[variant]};
  margin-bottom: 0.25em;
`;

export default StyledH1;
