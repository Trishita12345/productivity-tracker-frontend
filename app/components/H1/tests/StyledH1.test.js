import React from 'react';
import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import StyledH1 from '../StyledH1';
import { VARIANTS } from '../../../utils/constants/variants';
import { COLORS } from '../../../utils/constants/colorCodes';
import 'jest-styled-components';

configure({ adapter: new Adapter() });

describe('<StyledH1 />', () => {
  const props = { variant: VARIANTS.PRIMARY };

  const ComponentRenderer = myProps => {
    const component = mount(<StyledH1 {...myProps} />);
    return component;
  };

  it('should render with Primary variants', () => {
    const component = ComponentRenderer(props);
    expect(component).toHaveStyleRule('color', COLORS.DARK_VIOLET);
    expect(component).toMatchSnapshot();
  });
  it('should render with Secondary variants', () => {
    const component = ComponentRenderer({
      ...props,
      variant: VARIANTS.SECONDARY,
    });
    expect(component).toHaveStyleRule('color', COLORS.WHITE);
    expect(component).toMatchSnapshot();
  });
  it('should render with Tertiary variants', () => {
    const component = ComponentRenderer({
      ...props,
      variant: VARIANTS.TERTIARY,
    });
    expect(component).toHaveStyleRule('color', COLORS.DARK_GREY);
    expect(component).toMatchSnapshot();
  });
});
