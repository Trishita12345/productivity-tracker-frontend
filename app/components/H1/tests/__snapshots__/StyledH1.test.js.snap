// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StyledH1 /> should render with Primary variants 1`] = `ReactWrapper {}`;

exports[`<StyledH1 /> should render with Secondary variants 1`] = `ReactWrapper {}`;

exports[`<StyledH1 /> should render with Tertiary variants 1`] = `ReactWrapper {}`;
